sdk/azcore/runtime/policy_retry_test.go

```diff
@@ -810,6 +810,47 @@ func TestRetryableRequestBodyWithCloser(t *testing.T) {
 	require.True(t, tr.closeCalled)
 }
 
+func TestRetryPolicySuccessWithRetryPreserveHeaders(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithStatusCode(http.StatusRequestTimeout))
+	srv.AppendResponse()
+	pl := exported.NewPipeline(srv, NewRetryPolicy(testRetryOptions()), exported.PolicyFunc(challengeLikePolicy))
+	req, err := NewRequest(context.Background(), http.MethodGet, srv.URL())
+	require.NoError(t, err)
+	body := newRewindTrackingBody("stuff")
+	require.NoError(t, req.SetBody(body, "text/plain"))
+	resp, err := pl.Do(req)
+	require.NoError(t, err)
+	require.EqualValues(t, http.StatusOK, resp.StatusCode)
+	require.EqualValues(t, 2, srv.Requests())
+	require.EqualValues(t, 1, body.rcount)
+	require.True(t, body.closed)
+}
+
+type reqBody struct {
+	body        io.ReadSeekCloser
+	contentType string
+}
+
+func challengeLikePolicy(req *policy.Request) (*http.Response, error) {
+	if req.Body() == nil {
+		return nil, errors.New("request body wasn't restored")
+	}
+	if req.Raw().Header.Get("content-type") != "text/plain" {
+		return nil, errors.New("content-type header wasn't restored")
+	}
+
+	if body := req.Body(); body != nil {
+		rb := reqBody{body, req.Raw().Header.Get("content-type")}
+		req.SetOperationValue(rb)
```

Comment:
My first thought was, the actual policy does this but it isn't necessary for this test. Now I'm wondering whether the OperationValue feature needs additional testing. The actual policy's logic is:
1. set: `req.SetOperationValue()`
2. send: `req.Next()`
3. get: `req.OperationValue()`

It appears every retry runs on its own clone of `req`, and a policy never gets `req` itself. So step 1 sets the value on an ephemeral clone; if there's a retry during step 2, is that value lost?

Reply:
I see. Then I'm back to wondering why this test sets the value 😆. Does it have a side effect?

Reply:
No, it's just copy/pasta from the key vault challenge policy.  I'll clean this up a bit.

Reply:
Done.